<a href="https://colab.research.google.com/github/azizahulhusnah/Pemrograman-Dasar/blob/main/LATIHAN.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [5]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

**Memuat Tabel**

In [6]:
ziza_df = pd.read_csv('/content/mahasiswa (1).csv')
#import file csv

**Load tabel**

In [7]:
ziza_df.head()
#melihat 5 data teratas

nim   nama           jurusan    umur nilai_akhir  kehadiran_pct  \
0  E0001   Aldi            Teknik     150        59.5           82.0   
1  E0002   Budi            teknik    20.0        84.5           99.0   
2  E0003  Citra            Energi      -5        56.2           78.0   
3  E0004   Dewi            Teknik  twenty        63.0           99.0   
4  E0005    Eka  Manajemen Energi    22.0         999           65.0   

  tanggal_daftar  
0     2024-05-05  
1     2024-02-14  
2     2024-04-29  
3     2024-08-02  
4     2024-03-22

Kode tersebut dimulai dengan melakukan import beberapa library penting, yaitu numpy, pandas, matplotlib.pyplot, dan seaborn. Library-library ini digunakan untuk analisis data dan visualisasi grafik. Setelah itu, program membaca file CSV yang berisi data mahasiswa menggunakan fungsi pd.read_csv(), lalu hasilnya disimpan dalam sebuah DataFrame bernama ziza_df. DataFrame ini merupakan bentuk tabel yang memudahkan proses analisis. Untuk memastikan bahwa data telah berhasil dimuat dengan benar, kode kemudian menampilkan lima baris pertama dari tabel menggunakan ziza_df.head(). Dari hasil tampilan data tersebut, terlihat beberapa kolom seperti nim, nama, jurusan, umur, nilai_akhir, kehadiran_pct, dan tanggal_daftar. Namun, pada sebagian data ditemukan nilai yang tidak valid, misalnya umur yang terlalu besar, nilai negatif, atau berupa teks seperti “twenty”, serta nilai akhir 999 yang tampaknya merupakan outlier. Secara keseluruhan, kode ini berfungsi untuk mempersiapkan data sebelum dilakukan pembersihan atau analisis lebih lanjut.

In [8]:
ziza_df.tail()
#melihat 5 data terakhir

nim   nama      jurusan  umur nilai_akhir  kehadiran_pct tanggal_daftar
73  E0057  Fauzi       Energi  23.0        58.2           76.0     2024-02-05
74  E0039   Mita       Energi  23.0        77.2           79.0     2024-01-04
75  E0069  Salsa  Informatika  23.0        63.4           86.0     2024-04-14
76  E0031   Eros  Informatika  23.0        68.0           66.0     2024-02-27
77  E0026   Zara  Informatika  24.0        90.4           82.0     2024-11-24

Kode ini digunakan untuk menampilkan 5 baris terakhir dari DataFrame ziza_df.
Fungsinya adalah untuk mengecek kondisi data di bagian akhir dataset—apakah ada data yang aneh, kosong, atau tidak konsisten.

Hasilnya pada gambar menampilkan baris indeks 73–77 yang menunjukkan data mahasiswa terakhir dalam tabel.

**Assessing Data**

In [9]:
ziza_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 78 entries, 0 to 77
Data columns (total 7 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   nim             78 non-null     object 
 1   nama            78 non-null     object 
 2   jurusan         76 non-null     object 
 3   umur            76 non-null     object 
 4   nilai_akhir     76 non-null     object 
 5   kehadiran_pct   76 non-null     float64
 6   tanggal_daftar  76 non-null     object 
dtypes: float64(1), object(6)
memory usage: 4.4+ KB


Kode ziza_df.info() digunakan untuk melihat ringkasan lengkap mengenai struktur DataFrame. Ketika kode ini dijalankan, pandas menampilkan informasi penting seperti jumlah baris pada dataset, daftar nama kolom, jumlah nilai yang tidak kosong pada setiap kolom, serta tipe data dari tiap kolom tersebut. Dari hasil yang ditampilkan, terlihat bahwa dataset memiliki total 78 baris dan 7 kolom. Selain itu, beberapa kolom seperti jurusan, umur, dan kehadiran_pct memiliki jumlah data non-null yang lebih sedikit, yang berarti ada nilai kosong atau data yang tidak valid. Tipe data pada beberapa kolom juga masih dalam bentuk object, padahal seharusnya kolom seperti umur idealnya bertipe numerik. Informasi ini sangat berguna sebagai langkah awal untuk mengevaluasi kualitas data dan mengetahui bagian mana yang perlu dibersihkan sebelum dilakukan analisis lebih lanjut.

dari hasil infor terdapat jumlah data yg tidak konsisten ada 2 kolom yg jumlah nya 28 sedangkan yg lainya hanya 76

In [10]:
ziza_df.isna().sum()

nim               0
nama              0
jurusan           2
umur              2
nilai_akhir       2
kehadiran_pct     2
tanggal_daftar    2
dtype: int64

Kode ziza_df.isna().sum() digunakan untuk menghitung jumlah nilai yang hilang (missing values atau NaN) pada setiap kolom dalam DataFrame. Fungsi isna() akan memeriksa setiap sel dan menandai apakah sel tersebut kosong, sedangkan sum() akan menjumlahkan total nilai kosong per kolom. Dari hasil output terlihat bahwa terdapat beberapa kolom yang memiliki nilai NaN, yaitu kolom jurusan, umur, nilai_akhir, kehadiran_pct, dan tanggal_daftar, masing-masing memiliki 2 data yang hilang. Sementara kolom nim dan nama tidak memiliki nilai kosong sama sekali. Informasi ini sangat penting karena menunjukkan bahwa terdapat ketidakkonsistenan pada data, terutama terlihat dari jumlah entri per kolom yang sebelumnya ditunjukkan oleh info(), di mana beberapa kolom berisi 76 data sementara dataset total memiliki 78 baris. Dengan mengetahui kolom mana saja yang memiliki missing values, kita dapat merencanakan langkah pembersihan data (data cleaning) berikutnya, seperti mengisi nilai kosong, menghapus baris yang tidak lengkap, atau melakukan imputasi sesuai kebutuhan analisis.

Terdapat 5 kolom yang NaN atau missing values

In [11]:
print("Jumlah duplikasi: ", ziza_df.duplicated().sum())

Jumlah duplikasi:  8


Kode print("Jumlah duplikasi: ", ziza_df.duplicated().sum()) digunakan untuk menghitung jumlah baris yang terduplikasi dalam DataFrame. Fungsi duplicated() akan memeriksa setiap baris dan menandai apakah baris tersebut sama persis dengan baris lain, sementara sum() akan menjumlahkan berapa banyak baris yang teridentifikasi sebagai duplikat. Dari output terlihat bahwa terdapat 8 data duplikasi, yang berarti terdapat baris-baris yang benar-benar identik dan perlu dibersihkan agar analisis data tidak bias atau tidak akurat.

Lalu terdapat juga 8 data duplikasi

**Cleaning Data**

Menghilangkan duplicate data dengan drop_duplicates()

In [12]:
ziza_df.drop_duplicates(inplace=True)

Kode tersebut digunakan untuk menghapus baris-baris yang terduplikasi di dalam DataFrame ziza_df. Fungsi drop_duplicates() bekerja dengan cara memeriksa setiap baris dan mencari apakah ada baris lain yang memiliki nilai yang sama persis pada seluruh kolom. Jika ditemukan baris yang identik, baris tersebut dianggap sebagai duplikat dan akan dihapus. Penggunaan parameter inplace=True berarti perubahan dilakukan langsung pada DataFrame asli tanpa perlu membuat variabel baru. Dengan kata lain, DataFrame ziza_df akan langsung diperbarui sehingga baris duplikat hilang secara permanen. Langkah ini penting dalam proses pembersihan data karena keberadaan duplikasi dapat menyebabkan analisis yang bias, perhitungan statistik yang tidak akurat, serta visualisasi yang menyesatkan. Setelah kode ini dijalankan, biasanya dilakukan pengecekan ulang menggunakan ziza_df.duplicated().sum() untuk memastikan bahwa

In [13]:
print("Jumlah duplikasi: ", ziza_df.duplicated().sum())

Jumlah duplikasi:  0


Sekarang data duplikat sudah selesai

Pada bagian Cleaning Data, digunakan perintah ziza_df.drop_duplicates(inplace=True) untuk menghapus baris duplikat tersebut. Parameter inplace=True memastikan bahwa perubahan dilakukan langsung pada DataFrame asli tanpa perlu membuat variabel baru. Setelah proses pembersihan dilakukan, kode kembali dihitung menggunakan ziza_df.duplicated().sum() dan hasilnya menunjukkan angka 0, yang berarti seluruh data duplikat telah berhasil dihapus. Dengan demikian, dataset kini lebih bersih dan siap untuk melalui tahap pembersihan berikutnya seperti memperbaiki missing values atau data yang tidak valid.

**Menangani missing value**

In [14]:
ziza_df.isna().sum()

nim               0
nama              0
jurusan           2
umur              2
nilai_akhir       2
kehadiran_pct     2
tanggal_daftar    2
dtype: int64

ode ziza_df.isna().sum() digunakan untuk mengevaluasi kembali jumlah nilai kosong atau missing values dalam DataFrame setelah proses pembersihan sebelumnya. Fungsi isna() akan memeriksa setiap sel pada seluruh kolom dan menandai nilai mana saja yang kosong, kemudian fungsi sum() akan menjumlahkan total nilai kosong pada masing-masing kolom. Dari hasil yang ditampilkan, terlihat bahwa masih ada lima kolom yang memiliki missing values, yaitu jurusan, umur, nilai_akhir, kehadiran_pct, dan tanggal_daftar, masing-masing memiliki 2 nilai yang hilang. Sementara kolom nim dan nama tidak memiliki nilai kosong. Informasi ini sangat penting untuk mengetahui bagian mana dari dataset yang masih perlu diperbaiki sebelum melanjutkan proses analisis. Missing values ini nantinya bisa ditangani dengan beberapa metode, seperti mengisi dengan rata-rata, median, modus, atau menghapus baris terkait, tergantung kebutuhan analisis dan karakteristik data.

In [15]:
ziza_df['jurusan'] = ziza_df['jurusan'].fillna("Pembangkit Energi")

#di sini untuk jurusan di isi dengan "Pembangkit Energi"

Kode ziza_df['jurusan'] = ziza_df['jurusan'].fillna("Pembangkit Energi") digunakan untuk menangani nilai kosong (missing values) pada kolom jurusan. Fungsi fillna() bekerja dengan cara mengganti semua nilai NaN dalam kolom tersebut dengan nilai yang ditentukan, yaitu "Pembangkit Energi". Dengan menuliskan hasilnya kembali ke ziza_df['jurusan'], maka perubahan tersebut langsung diterapkan pada kolom yang sama. Langkah ini penting karena kolom jurusan tidak boleh dibiarkan kosong—setiap mahasiswa harus memiliki jurusan yang jelas—sehingga mengisinya dengan nilai yang konsisten dapat membantu menjaga kualitas data dan mempermudah proses analisis selanjutnya. Teknik pengisian ini disebut imputation, dan digunakan terutama ketika jumlah nilai kosong relatif sedikit dan dapat digantikan dengan nilai yang logis atau relevan.

In [16]:
ziza_df.isna().sum()

nim               0
nama              0
jurusan           0
umur              2
nilai_akhir       2
kehadiran_pct     2
tanggal_daftar    2
dtype: int64

ode tersebut digunakan untuk memeriksa kembali jumlah nilai kosong (missing values atau NaN) pada setiap kolom dalam DataFrame ziza_df. Fungsi isna() akan mengecek seluruh sel pada tabel dan menandai mana yang bernilai kosong, sementara sum() menjumlahkan total nilai kosong di masing-masing kolom. Hasil dari perintah ini menunjukkan berapa banyak data yang masih belum terisi setelah dilakukan proses pembersihan sebelumnya, seperti penghapusan duplikasi atau pengisian nilai jurusan. Pemeriksaan ini penting untuk memastikan bahwa tidak ada lagi missing values yang tertinggal, sehingga dataset benar-benar siap untuk tahap analisis berikutnya tanpa risiko munculnya error atau hasil yang bias.

In [17]:
ziza_df['umur'] = ziza_df['umur'].fillna(23)

Kode tersebut digunakan untuk menangani nilai kosong (missing values) pada kolom umur di DataFrame ziza_df. Fungsi fillna(23) berarti semua nilai umur yang sebelumnya berstatus NaN akan diganti atau diisi dengan angka 23. Angka 23 dipilih sebagai nilai imputasi, biasanya karena merupakan nilai yang dianggap paling masuk akal—misalnya usia rata-rata mahasiswa dalam dataset tersebut. Dengan menyimpan hasilnya kembali ke ziza_df['umur'], maka kolom umur langsung diperbarui dan tidak lagi memiliki nilai kosong. Langkah ini penting untuk memastikan bahwa seluruh nilai dalam kolom umur bersifat numerik dan lengkap, sehingga tidak menimbulkan error pada proses analisis berikutnya seperti perhitungan statistik, visualisasi, atau machine learning.

In [18]:
ziza_df.isna().sum()

nim               0
nama              0
jurusan           0
umur              0
nilai_akhir       2
kehadiran_pct     2
tanggal_daftar    2
dtype: int64

Kode ini digunakan untuk mengecek kembali jumlah nilai kosong (missing values) di setiap kolom setelah dilakukan proses pengisian nilai pada kolom tertentu, seperti jurusan dan umur. Fungsi isna() akan memeriksa seluruh elemen dalam DataFrame dan menandai mana saja yang bernilai NaN, lalu sum() menjumlahkan total nilai kosong tersebut per kolom. Dengan menjalankan perintah ini, kamu dapat memastikan apakah proses imputasi sebelumnya sudah berhasil menghilangkan missing values, atau apakah masih ada kolom lain yang memerlukan penanganan tambahan. Pemeriksaan ini merupakan langkah penting dalam proses data cleaning untuk memastikan dataset benar-benar bersih dan siap digunakan untuk analisis lebih lanjut.

In [19]:
ziza_df['nilai_akhir'] = pd.to_numeric(ziza_df['nilai_akhir'], errors='coerce') #konversi tipe data menjadi numeric yaitu float

ziza_df['nilai_akhir'].mean() #mecari rata-rata

np.float64(88.15384615384616)

Kode ziza_df['nilai_akhir'] = pd.to_numeric(ziza_df['nilai_akhir'], errors='coerce') digunakan untuk mengubah kolom nilai_akhir menjadi tipe data numerik, khususnya float. Pada beberapa dataset, kolom nilai bisa saja berisi karakter non‐angka seperti teks, simbol, atau angka yang ditulis tidak sesuai format. Dengan menggunakan pd.to_numeric(), semua nilai yang valid akan dikonversi menjadi angka, sedangkan nilai yang tidak dapat diubah akan otomatis diganti menjadi NaN karena parameter errors='coerce'. Hal ini sangat penting agar kolom tersebut bisa digunakan untuk perhitungan statistik tanpa menimbulkan error.

Setelah kolom berhasil dikonversi menjadi numerik, kode ziza_df['nilai_akhir'].mean() digunakan untuk menghitung nilai rata-rata dari seluruh data pada kolom nilai_akhir. Fungsi mean() akan menjumlahkan semua nilai kemudian membaginya dengan jumlah data yang valid (mengabaikan NaN). Perhitungan ini bermanfaat untuk memahami gambaran umum performa mahasiswa berdasarkan nilai akhir mereka, terutama setelah data sudah dibersihkan dan siap dianalisis.

In [20]:
ziza_df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 70 entries, 0 to 69
Data columns (total 7 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   nim             70 non-null     object 
 1   nama            70 non-null     object 
 2   jurusan         70 non-null     object 
 3   umur            70 non-null     object 
 4   nilai_akhir     65 non-null     float64
 5   kehadiran_pct   68 non-null     float64
 6   tanggal_daftar  68 non-null     object 
dtypes: float64(2), object(5)
memory usage: 4.4+ KB


Kode ziza_df.info() digunakan untuk menampilkan ringkasan struktur lengkap dari DataFrame setelah berbagai proses pembersihan data dilakukan, seperti menghapus duplikasi, mengisi nilai kosong, dan mengonversi tipe data. Fungsi ini menampilkan jumlah total baris data, nama setiap kolom, jumlah nilai yang tidak kosong (non-null), serta tipe data masing-masing kolom. Dengan menjalankan kode ini, kamu dapat melihat apakah perubahan yang sudah dilakukan sebelumnya berhasil, misalnya apakah kolom nilai_akhir sudah berubah menjadi tipe numerik, apakah kolom jurusan dan umur sudah tidak memiliki missing values, dan apakah tipe data seluruh kolom sudah sesuai untuk dianalisis lebih lanjut. Pemeriksaan ini penting sebagai tahap akhir dari proses data cleaning untuk memastikan bahwa dataset berada dalam kondisi yang benar-benar siap digunakan untuk analisis statistik, visualisasi, atau pemodelan.

Jika kita lihat di sini tipe dara pada nilai_akhir sudah berubah menjadi float

In [21]:
ziza_df['nilai_akhir'] = ziza_df['nilai_akhir'].fillna(88)
#sekarang isi missing value pada nilai_akhir dengan nilai rata-rata yg sudah di dapat sebelumnya

Kode ziza_df['nilai_akhir'] = ziza_df['nilai_akhir'].fillna(88) digunakan untuk mengatasi nilai kosong (missing values) pada kolom nilai_akhir. Setelah sebelumnya kolom ini dikonversi menjadi tipe numerik dan dihitung nilai rata-ratanya, angka 88 diperoleh sebagai nilai rata-rata tersebut. Dengan menggunakan fungsi fillna(88), semua sel yang bernilai NaN pada kolom nilai_akhir akan digantikan oleh angka 88. Langkah ini memastikan bahwa tidak ada lagi nilai kosong yang dapat mengganggu proses analisis, perhitungan statistik, atau pembuatan grafik nanti. Mengisi missing value dengan rata-rata adalah metode imputasi yang umum digunakan ketika data yang hilang jumlahnya sedikit dan dianggap tidak memiliki pola tertentu. Dengan demikian, dataset menjadi lebih bersih, konsisten, dan siap digunakan untuk analisis lebih lanjut.

In [22]:
ziza_df.isna().sum()
#cek hasilnya

nim               0
nama              0
jurusan           0
umur              0
nilai_akhir       0
kehadiran_pct     2
tanggal_daftar    2
dtype: int64

ode tersebut digunakan untuk memeriksa kembali jumlah nilai kosong (missing values) pada setiap kolom setelah proses pengisian nilai pada kolom nilai_akhir dilakukan. Fungsi isna() akan mengecek semua sel dalam DataFrame dan menandai sel mana yang bernilai NaN, kemudian sum() akan menghitung total nilai kosong per kolom. Dengan menjalankan perintah ini, kamu bisa memastikan apakah pengisian missing value pada kolom nilai_akhir menggunakan nilai rata-rata (88) sudah berhasil dan apakah masih ada kolom lain yang masih memiliki nilai kosong. Tahap pengecekan ini sangat penting dalam proses data cleaning, karena memastikan bahwa dataset sudah bersih dan tidak lagi memiliki nilai hilang yang dapat menyebabkan error atau analisis yang tidak akurat pada langkah-langkah berikutnya.

**Lakukan untuk kolom Kehadiran**

In [23]:
ziza_df['kehadiran_pct'] = pd.to_numeric(ziza_df['kehadiran_pct'], errors='coerce') #konversi tipe data menjadi numeric yaitu float

ziza_df['kehadiran_pct'].mean() #mecari rata-rata

np.float64(82.13235294117646)

Kode ziza_df['kehadiran_pct'] = pd.to_numeric(ziza_df['kehadiran_pct'], errors='coerce') digunakan untuk mengubah kolom kehadiran_pct menjadi tipe data numerik (float). Pada dataset, kolom ini mungkin berisi data yang tidak sepenuhnya berbentuk angka, seperti teks, simbol, atau format persentase yang tidak sesuai. Fungsi pd.to_numeric() akan mencoba mengonversi semua nilai dalam kolom tersebut menjadi angka. Jika ada nilai yang tidak bisa diubah menjadi angka, fungsi ini akan menggantinya dengan nilai NaN karena parameter errors='coerce'. Langkah ini penting agar kolom kehadiran dapat diproses dalam perhitungan statistik tanpa menimbulkan kesalahan.
Setelah perubahan tipe data berhasil dilakukan, kode ziza_df['kehadiran_pct'].mean() digunakan untuk menghitung rata-rata kehadiran mahasiswa dalam dataset. Fungsi mean() akan mengabaikan nilai NaN dan menghitung rata-rata berdasarkan nilai yang valid saja. Hasil rata-rata ini nantinya dapat digunakan sebagai nilai imputasi jika kamu ingin mengganti missing value di kolom kehadiran_pct dengan nilai yang representatif. Proses ini merupakan bagian penting dalam pembersihan dan persiapan data sebelum masuk ke tahap analisis lebih lanjut.

In [24]:
ziza_df['kehadiran_pct'] = ziza_df['kehadiran_pct'].fillna(82)
#sekarang isi missing value pada nilai_akhir dengan nilai rata-rata yg sudah di dapat sebelumnya

Kode ziza_df['kehadiran_pct'] = ziza_df['kehadiran_pct'].fillna(82) digunakan untuk mengisi nilai kosong (missing values) pada kolom kehadiran_pct setelah sebelumnya kolom tersebut dikonversi menjadi numerik dan dihitung nilai rata-ratanya. Angka 82 merupakan nilai rata-rata kehadiran yang diperoleh dari perhitungan sebelumnya menggunakan fungsi mean(). Dengan menggunakan fillna(82), semua nilai NaN pada kolom ini akan digantikan oleh angka 82 sehingga kolom menjadi lengkap tanpa ada data yang hilang. Langkah ini penting karena nilai kosong dapat mengganggu proses analisis, sehingga pengisian menggunakan rata-rata adalah metode imputasi yang umum dan efektif. Setelah proses ini dilakukan, kolom kehadiran_pct siap digunakan untuk analisis statistik maupun visualisasi tanpa menimbulkan error.

In [25]:
ziza_df.isna().sum()
#cek hasilnya

nim               0
nama              0
jurusan           0
umur              0
nilai_akhir       0
kehadiran_pct     0
tanggal_daftar    2
dtype: int64

Kode tersebut digunakan untuk melakukan pengecekan ulang terhadap jumlah nilai kosong (missing values) pada setiap kolom setelah proses pengisian nilai pada kolom kehadiran_pct dilakukan. Fungsi isna() akan memeriksa seluruh sel di dalam DataFrame dan menandai mana saja yang masih bernilai NaN, kemudian fungsi sum() akan menjumlahkannya per kolom sehingga kamu dapat melihat apakah masih ada data yang belum terisi. Dengan menjalankan perintah ini, kamu dapat memastikan bahwa pengisian missing value menggunakan nilai rata-rata sebelumnya sudah berjalan dengan benar dan bahwa kolom kehadiran_pct kini tidak lagi memiliki nilai kosong. Pengecekan ini merupakan langkah penting dalam proses data cleaning, karena memastikan bahwa seluruh dataset telah bersih dan siap untuk dianalisis tanpa adanya masalah data hilang yang dapat mengganggu hasil analisis.

Lakukan untuk kolom tanggal daftar

In [26]:
ziza_df['tanggal_daftar'] = ziza_df['tanggal_daftar'].fillna('2024-03-24')

# karena tanggal lagir di sini bertipe string jadi langsung masukan saja tidak perlu di cari rata-rata nya atau konversi ke numerik

Kode ziza_df['tanggal_daftar'] = ziza_df['tanggal_daftar'].fillna('2024-03-24') digunakan untuk mengisi nilai kosong (missing values) pada kolom tanggal_daftar. Kolom ini berisi data tanggal yang masih disimpan dalam bentuk string atau object, sehingga tidak perlu dilakukan konversi ke tipe numerik maupun perhitungan nilai rata-rata seperti pada kolom angka. Karena sifat data tanggal tidak bisa dirata-ratakan, nilai NaN diganti secara langsung dengan satu tanggal yang dianggap sesuai, yaitu "2024-03-24". Dengan menggunakan fillna(), setiap sel yang kosong otomatis diisi dengan tanggal tersebut sehingga kolom tanggal_daftar menjadi lengkap tanpa adanya data kosong. Langkah ini penting untuk memastikan dataset lebih rapi dan konsisten sebelum masuk ke tahap analisis selanjutnya.

In [27]:
ziza_df.isna().sum()

nim               0
nama              0
jurusan           0
umur              0
nilai_akhir       0
kehadiran_pct     0
tanggal_daftar    0
dtype: int64

Kode tersebut digunakan untuk melakukan pengecekan terakhir terhadap jumlah nilai kosong (missing values) di setiap kolom setelah seluruh proses pengisian data (imputasi) dilakukan. Fungsi isna() akan memeriksa semua sel di dalam DataFrame dan menandai mana yang masih bernilai NaN, kemudian fungsi sum() akan menghitung total nilai kosong pada masing-masing kolom. Dengan menjalankan perintah ini, kamu dapat memastikan bahwa semua kolom seperti jurusan, umur, nilai_akhir, kehadiran_pct, dan tanggal_daftar sudah terisi penuh dan tidak lagi memiliki missing values. Pengecekan ini merupakan langkah penting sebagai validasi akhir proses data cleaning, memastikan bahwa dataset sudah benar-benar bersih dan siap digunakan untuk analisis, visualisasi, atau pemodelan data tanpa hambatan dari data yang hilang.

Sekarang tidak ada lagi data missing atau NaN

**Menangani inaccurate value**


In [28]:
ziza_df.jurusan.value_counts()

jurusan
Energi               21
Informatika          17
Teknik               12
Manajemen Energi     11
Pembangkit Energi     2
teknik                1
 energi               1
ManajemenEnergi       1
EnErGi                1
Teknik                1
ENG                   1
ENERGI                1
Name: count, dtype: int64

Kode tersebut digunakan untuk melihat jumlah kemunculan setiap jenis jurusan yang ada di dalam kolom jurusan pada DataFrame. Fungsi value_counts() akan menghitung seberapa sering setiap kategori muncul dan menampilkannya dari yang paling banyak hingga paling sedikit. Langkah ini penting ketika menangani inaccurate value atau nilai yang tidak konsisten, karena melalui hasil perhitungan ini kita dapat melihat apakah ada penulisan jurusan yang berbeda tetapi sebenarnya bermakna sama, misalnya “Teknik Mesin” dan “teknik mesin”, atau jurusan yang salah ketik maupun kategori yang seharusnya tidak ada. Dengan mengetahui distribusi ini, kita dapat melakukan pembersihan lebih lanjut seperti menyatukan format penamaan, memperbaiki kesalahan penulisan, atau menghapus kategori yang tidak valid sehingga data menjadi lebih akurat, rapi, dan siap untuk dianalisis.

In [29]:
ziza_df.umur.value_counts()

umur
23.0          15
25.0          10
20.0           9
24.0           9
22.0           8
19.0           7
21.0           4
23             2
150            1
twenty         1
-5             1
120            1
NaN_string     1
21years        1
Name: count, dtype: int64

Kode tersebut digunakan untuk melihat distribusi atau jumlah kemunculan setiap nilai pada kolom umur di dalam DataFrame. Dengan memanggil value_counts(), pandas akan menghitung berapa banyak mahasiswa yang memiliki umur tertentu dan menampilkannya secara terurut dari yang paling sering muncul hingga yang paling jarang. Langkah ini sangat penting ketika menangani inaccurate value atau nilai yang tidak valid pada kolom umur, misalnya umur yang terlalu besar seperti 150, umur negatif, atau nilai berupa teks seperti “twenty”. Dengan melihat hasil dari value_counts(), kita dapat dengan mudah mengidentifikasi nilai-nilai umur yang tidak wajar sehingga dapat dibersihkan, diganti, atau dikonversi sesuai kebutuhan. Analisis ini membantu memastikan bahwa kolom umur bersifat konsisten dan akurat sebelum dilakukan proses analisis statistik atau visualisasi data.

Terdapat 2 kolom yang datanya tidak akurat atau  inaccurate value


In [30]:
ziza_df['jurusan'] = ziza_df['jurusan'].str.lower().str.strip()
#ubah menjadi huruf kecil

Kode tersebut digunakan untuk membersihkan nilai pada kolom jurusan dengan cara menyeragamkan penulisannya. Fungsi .str.lower() mengubah seluruh teks dalam kolom jurusan menjadi huruf kecil, sehingga perbedaan penulisan seperti “Teknik Mesin”, “teknik mesin”, atau “TEKNIK MESIN” akan dianggap sama dan lebih mudah diproses. Sementara itu, fungsi .str.strip() digunakan untuk menghapus spasi berlebih di awal atau akhir teks yang sering tidak terlihat namun dapat menyebabkan nilai dianggap berbeda. Langkah normalisasi ini sangat penting ketika menangani inaccurate value, karena data kategori harus memiliki format yang konsisten agar proses analisis seperti pengelompokan, penghitungan frekuensi, atau visualisasi tidak menghasilkan hasil yang salah akibat perbedaan penulisan yang sebenarnya hanya variasi kecil. Dengan pembersihan ini, kolom jurusan menjadi lebih rapi, seragam, dan siap digunakan untuk analisis lanjutan.

In [31]:
ziza_df.jurusan.value_counts()

jurusan
energi               24
informatika          17
teknik               14
manajemen energi     11
pembangkit energi     2
manajemenenergi       1
eng                   1
Name: count, dtype: int64

Kode ini digunakan untuk menampilkan kembali jumlah kemunculan setiap kategori pada kolom jurusan setelah proses pembersihan data dilakukan, seperti mengubah huruf menjadi lowercase dan menghapus spasi menggunakan .str.lower().str.strip(). Dengan memanggil value_counts(), kita bisa melihat apakah penulisan jurusan kini sudah seragam dan tidak lagi memiliki variasi yang tidak konsisten, misalnya perbedaan antara “Teknik Mesin” dan “teknik mesin”. Hasil dari kode ini membantu memastikan bahwa proses normalisasi data pada kolom jurusan berhasil, sehingga kategori yang sebelumnya terpecah karena perbedaan penulisan kini sudah tergabung dalam satu kategori yang sama. Pemeriksaan ini penting untuk memastikan bahwa analisis data selanjutnya, seperti penghitungan frekuensi atau visualisasi, akan menghasilkan informasi yang akurat dan tidak bias akibat variasi penulisan yang tidak diperlukan.

In [32]:
#untuk memudahkan replacing maka di buatkan mapping jurusan sesuai dengan yg di inginkan, dengan kata-kata yg mirip di masukan ke jurusan yg seusai

mapping_jurusan = {
    "energi": "Pembangkit Energi",
    " energi ": "Pembangkit Energi",
    "ENERGI": "Pembangkit Energi",
    "EnErGi": "Pembangkit Energi",
    "eng": "Pembangkit Energi",
    "teknik": "Pembangkit Energi",
    "Teknik ": "Pembangkit Energi",
    "teknik ": "Pembangkit Energi",
    "manajemenenergi": "Manajemen Energi",
    "manajemen energi": "Manajemen Energi"
}

Mapping jurusan tersebut digunakan untuk menangani nilai-nilai yang tidak konsisten atau tidak akurat pada kolom jurusan. Ketidakkonsistenan biasanya muncul karena perbedaan penulisan, seperti huruf besar-kecil, spasi berlebih, atau singkatan yang tidak baku. Dengan membuat dictionary mapping_jurusan, kamu menentukan standar penulisan jurusan yang benar, misalnya menyamakan berbagai variasi kata seperti "energi", "ENERGI", "eng", atau "teknik" menjadi kategori yang seragam yaitu "Pembangkit Energi". Begitu juga dengan variasi "manajemenenergi" dan "manajemen energi" disamakan menjadi "Manajemen Energi". Pendekatan ini memudahkan proses replacing, karena setiap nilai yang mirip dapat langsung dipetakan ke kategori yang sesuai hanya dengan satu langkah .replace(mapping_jurusan). Dengan cara ini, data menjadi lebih bersih, rapi, dan siap digunakan untuk analisis tanpa terpecah menjadi kategori-kategori kecil akibat perbedaan penulisan.

In [33]:
ziza_df["jurusan"] = ziza_df["jurusan"].replace(mapping_jurusan)

berfungsi untuk menggantikan semua nilai pada kolom jurusan yang cocok dengan keys di dalam mapping_jurusan menjadi nilai standar yang sudah kamu tentukan.

Karena kamu sebelumnya sudah membuat dictionary berisi berbagai variasi penulisan jurusan, perintah .replace(mapping_jurusan) akan secara otomatis mencari nilai-nilai tersebut di kolom jurusan dan menggantinya dengan format yang benar, misalnya mengubah "energi", "ENERGI", "eng", atau "teknik" menjadi "Pembangkit Energi", begitu juga nilai lain seperti "manajemenenergi" menjadi "Manajemen Energi".

In [34]:
ziza_df["jurusan"] = ziza_df["jurusan"].str.title()

#kemablikan menjadi huruf besar

berfungsi untuk mengubah setiap kata pada kolom jurusan menjadi format Title Case, yaitu huruf pertama setiap kata menjadi huruf besar dan sisanya huruf kecil.

In [35]:
ziza_df.jurusan.value_counts()

jurusan
Pembangkit Energi    41
Informatika          17
Manajemen Energi     12
Name: count, dtype: int64

igunakan untuk melihat jumlah kemunculan setiap kategori dalam kolom jurusan setelah seluruh proses pembersihan data dilakukan. Dengan fungsi ini, kamu bisa mengetahui apakah nilai-nilai jurusan sudah konsisten, apakah masih ada kategori yang salah tulis, serta memastikan bahwa proses normalisasi seperti mengubah huruf kecil, melakukan mapping, dan mengembalikan ke format title case sudah bekerja dengan benar. Hasil yang ditampilkan berupa daftar jurusan beserta jumlah mahasiswa di masing-masing kategori. Ini membantu memverifikasi bahwa data sudah rapi dan siap digunakan untuk analisis selanjutnya.

In [36]:
print("Hasil setelah normalisasi dan pemetaan:")
print(ziza_df['jurusan'].value_counts())

Hasil setelah normalisasi dan pemetaan:
jurusan
Pembangkit Energi    41
Informatika          17
Manajemen Energi     12
Name: count, dtype: int64


Kode tersebut digunakan untuk menampilkan hasil akhir dari proses normalisasi dan pemetaan data pada kolom jurusan. Baris pertama mencetak teks penjelas agar output lebih mudah dipahami. Baris kedua menampilkan jumlah kemunculan setiap kategori jurusan setelah seluruh proses pembersihan dilakukan—mulai dari mengubah huruf menjadi lowercase, menghapus spasi berlebih, mengganti nilai-nilai yang tidak konsisten menggunakan dictionary mapping, hingga mengembalikan format huruf menjadi title case. Dengan memanggil value_counts(), kamu dapat melihat distribusi jurusan yang sudah seragam dan konsisten, sehingga mudah memastikan apakah proses normalisasi berhasil dan tidak ada lagi nilai jurusan yang salah atau beragam bentuk penulisannya.

Sekarang untuk inaccurate value pada kolom jurusan sudah selesai dan tidak da yg type atau tidak terkelompok dengan benar

**Selanjutnya lakuakan untuk kolom nilai**

In [37]:
ziza_df.umur.value_counts()

umur
23.0          15
25.0          10
20.0           9
24.0           9
22.0           8
19.0           7
21.0           4
23             2
150            1
twenty         1
-5             1
120            1
NaN_string     1
21years        1
Name: count, dtype: int64

Perintah tersebut digunakan untuk menghitung berapa banyak kemunculan masing-masing nilai pada kolom umur di dalam DataFrame ziza_df. Dengan menjalankan value_counts(), kamu dapat melihat distribusi usia para responden atau peserta setelah proses pembersihan data dilakukan—misalnya setelah mengisi nilai yang kosong dengan angka tertentu. Fungsi ini membantu mendeteksi apakah ada umur yang tidak wajar, apakah ada nilai yang muncul terlalu sering, atau apakah data umur sudah konsisten. Hasil akhirnya berupa daftar nilai umur dan jumlah kemunculannya, sehingga memudahkanmu memeriksa apakah data tersebut sudah siap digunakan untuk analisis berikutnya.

di sini terlihat beberapa data yg tidak konsisten seperti (150, -5, twenty, 120,NaN_string, 21years

Selanjutnya lakukan seperti pada kolom jurusan , bisa di lakukan mapping atau di ganti satu persatu, karena di sini jumlah data yg inkonsisten hanya satum jadi bisa di replace saja

In [38]:
ziza_df['umur'] = ziza_df['umur'].replace('twenty', 20)

#bisa di lakuakn replace satu persatu seperti ini
# dari sebelumnya twenty menjadi 20

Kode ziza_df['umur'] = ziza_df['umur'].replace('twenty', 20) digunakan untuk memperbaiki nilai yang tidak akurat pada kolom umur, khususnya nilai yang ditulis dalam bentuk teks seperti "twenty". Dengan menggunakan fungsi replace(), nilai teks tersebut diganti menjadi angka numerik 20, sehingga format data pada kolom umur menjadi lebih konsisten dan dapat digunakan untuk analisis numerik selanjutnya. Cara ini termasuk metode manual replacing, yaitu mengganti nilai tidak valid satu per satu. Pendekatan seperti ini bermanfaat jika jumlah nilai yang salah tidak terlalu banyak dan mudah diidentifikasi. Setelah penggantian dilakukan, nilai umur tidak lagi berisi teks sehingga meminimalkan error dalam perhitungan statistik data.

In [39]:
print(ziza_df['umur'].value_counts())

umur
23.0          15
25.0          10
20.0           9
24.0           9
22.0           8
19.0           7
21.0           4
23             2
150            1
20             1
-5             1
120            1
NaN_string     1
21years        1
Name: count, dtype: int64


Perintah ini digunakan untuk menampilkan jumlah kemunculan setiap nilai umur dalam kolom umur secara langsung. Dengan memanggil value_counts(), kamu bisa melihat distribusi umur setelah proses pembersihan dilakukan, misalnya setelah mengganti nilai seperti "twenty" menjadi angka 20 atau setelah mengisi nilai yang hilang. Fungsi ini membantu memastikan bahwa data umur sudah konsisten, tidak ada nilai aneh, dan seluruh elemen sudah berada dalam format numerik. Hasil yang ditampilkan membuatmu dapat memeriksa apakah masih ada nilai yang perlu diperbaiki sebelum melakukan analisis lebih lanjut.

Bisa di lakuakn replace satu persatu seperti di atas atau mapping kembali seperti jurusan sebelumnya

In [40]:
"""mapping_umur = {
    'twenty': 20,
    '21years': 21,
    'NaN_string': np.nan,
    -5: np.nan,
    120: np.nan,
    150: np.nan
}
"""
#untuk nilai yang NaN_string, -5, 120, 150 bisa langsung di ganti dengan rata-rata umur atau di jadikan missing value, tetapi jika di jadikan missing value maka setelah ini anda akan mengganti missing value tersebut, jadi sebelum itu lebih baik  di cari rata-rata nya terlebih dahulu lalu di ganti pada mappingan ini

"mapping_umur = {\n    'twenty': 20,\n    '21years': 21,\n    'NaN_string': np.nan,\n    -5: np.nan,\n    120: np.nan,\n    150: np.nan\n}\n"

Potongan kode mapping yang ditulis dalam komentar adalah contoh cara membuat sebuah dictionary bernama mapping_umur yang berfungsi untuk memperbaiki nilai–nilai tidak akurat pada kolom umur. Dalam dictionary tersebut, beberapa nilai yang salah format seperti "twenty" dan "21years" diubah menjadi angka sebenarnya, yaitu 20 dan 21, agar konsisten dengan format numerik. Selain itu, nilai–nilai yang tidak masuk akal atau tidak valid seperti "NaN_string", –5, 120, dan 150 diarahkan untuk diganti menjadi np.nan, sehingga dianggap sebagai missing value. Nilai umur seperti –5 atau 150 memang tidak realistis sehingga perlu dikonversi menjadi NaN agar bisa ditangani dalam tahap pembersihan berikutnya. Komentar pada kode tersebut menjelaskan bahwa jika nilai–nilai ini dijadikan NaN, maka setelahnya kamu perlu mengganti NaN tersebut, misalnya menggunakan nilai rata-rata umur. Karena itu, sebelum membuat mapping, lebih baik menghitung rata-rata umur terlebih dahulu sehingga nilai–nilai yang tidak valid bisa langsung diganti dengan angka yang benar dan tidak perlu melalui langkah pengisian NaN lagi. Ini adalah proses penting untuk memastikan bahwa kolom umur menjadi bersih, akurat, dan siap digunakan untuk analisis lebih lanjut.

In [41]:
ziza_df['umur'] = pd.to_numeric(ziza_df['umur'], errors='coerce')
ziza_df['umur'].mean()

np.float64(25.220588235294116)

Kode tersebut digunakan untuk memastikan bahwa seluruh nilai pada kolom umur berada dalam format numerik. Perintah pd.to_numeric() berfungsi mengonversi setiap nilai menjadi angka. Jika masih terdapat nilai yang tidak bisa diubah menjadi angka—misalnya teks seperti "twenty", "21years", atau nilai yang salah—maka dengan opsi errors='coerce', nilai tersebut otomatis diubah menjadi NaN. Langkah ini sangat penting untuk membersihkan data sehingga kolom umur hanya berisi nilai valid yang dapat dihitung. Setelah proses konversi selesai, baris kedua ziza_df['umur'].mean() digunakan untuk menghitung rata-rata umur dari seluruh data yang valid. Nilai NaN yang muncul akibat konversi otomatis akan diabaikan oleh fungsi mean, sehingga hasil rata-rata tetap akurat. Proses ini merupakan bagian penting dalam pembersihan data sebelum melakukan penggantian missing value atau analisis statistik berikutnya.

Sekarang setelah di konversi menjadi numerik maka di dapat nilai rata-rata, dan selanjutnya isikan pada mappinganya

In [42]:
mapping_umur = {
    'twenty': 20,
    '21years': 21,
    'NaN_string': 25,
    -5: 25,
    120: 25,
    150: 25
}

Dictionary mapping_umur ini digunakan untuk memperbaiki nilai–nilai yang tidak akurat atau tidak valid pada kolom umur di DataFrame. Dalam mapping tersebut, beberapa nilai umur yang tertulis dalam format tidak benar, seperti "twenty" dan "21years", diubah menjadi nilai numerik sebenarnya yaitu 20 dan 21, agar data konsisten dan dapat dihitung secara statistik. Selain itu, nilai–nilai yang tidak masuk akal atau dianggap tidak valid seperti "NaN_string", –5, 120, dan 150 diganti dengan angka 25, yang di sini digunakan sebagai nilai pengganti atau rata-rata umur yang telah dihitung sebelumnya. Dengan memberikan nilai pengganti langsung dalam mapping, kamu tidak perlu lagi melakukan proses pengisian NaN setelahnya, sehingga pembersihan data menjadi lebih efisien. Mapping ini memastikan bahwa seluruh data umur menjadi bersih, realistis, dan siap digunakan untuk analisis selanjutnya tanpa gangguan dari nilai yang tidak sesuai.

In [43]:
ziza_df['umur'] = ziza_df['umur'].replace(mapping_umur)

Kode ini digunakan untuk mengganti nilai–nilai pada kolom umur berdasarkan dictionary mapping_umur yang telah dibuat sebelumnya. Dengan menggunakan fungsi replace(), setiap nilai yang cocok dengan key pada mapping akan diganti dengan nilai barunya (value). Misalnya, teks seperti "twenty" atau "21years" akan diganti menjadi angka 20 dan 21, sedangkan nilai yang tidak valid seperti "NaN_string", –5, 120, atau 150 akan diganti dengan angka 25 sesuai hasil rata-rata yang telah dihitung sebelumnya. Proses ini memastikan bahwa seluruh data pada kolom umur menjadi lebih konsisten, realistis, dan bebas dari nilai yang tidak masuk akal. Menggunakan mapping juga membuat pembersihan data lebih cepat dan efisien karena semua penggantian dapat dilakukan sekaligus hanya dalam satu perintah. Setelah langkah ini, kolom umur sudah siap digunakan untuk analisis statistik berikutnya.

In [44]:
print("Hasil setelah pemetaan:")
print(ziza_df['umur'].value_counts())
print(f"\nTipe data kolom 'umur' sekarang: {ziza_df['umur'].dtype}")

Hasil setelah pemetaan:
umur
23.0    17
25.0    13
20.0    10
24.0     9
22.0     8
19.0     7
21.0     4
Name: count, dtype: int64

Tipe data kolom 'umur' sekarang: float64


Kode ini digunakan untuk menampilkan hasil akhir setelah proses pemetaan nilai pada kolom umur selesai dilakukan. Baris pertama mencetak teks “Hasil setelah pemetaan:” sebagai penanda agar output lebih mudah dipahami. Baris kedua menggunakan value_counts() untuk menampilkan distribusi jumlah setiap nilai umur setelah semua nilai yang tidak valid atau tidak konsisten diganti menggunakan dictionary mapping_umur. Dengan cara ini, kamu dapat melihat apakah pemetaan sudah berhasil dan apakah data umur sudah tampak bersih serta realistis.

Selanjutnya, baris ketiga menampilkan tipe data dari kolom umur menggunakan dtype. Informasi ini penting karena memastikan apakah seluruh nilai benar-benar telah berubah menjadi tipe numerik, sehingga kolom tersebut siap digunakan untuk analisis statistik. Jika hasilnya adalah int64 atau float64, berarti proses konversi dan pembersihan telah berhasil. Kode ini membantu memverifikasi bahwa data akhir sudah rapi dan tidak lagi mengandung nilai yang bermasalah.

Sekarang kolom umur sudah tidak ada lagi data yg inaccruate,tetapi harus di pastikan lagi seperti di bawah ini

In [45]:
ziza_df.isna().sum()
#ternyata terdapat 2 nilai yang NaN pada kolom umur

nim               0
nama              0
jurusan           0
umur              2
nilai_akhir       0
kehadiran_pct     0
tanggal_daftar    0
dtype: int64

Perintah ziza_df.isna().sum() digunakan untuk menghitung jumlah nilai NaN atau missing value pada setiap kolom dalam DataFrame. Fungsi isna() akan menghasilkan nilai True untuk setiap sel yang berisi NaN, kemudian sum() akan menjumlahkan berapa banyak True tersebut di tiap kolom. Dari hasil pengecekan ini, kamu mengetahui bahwa kolom umur masih memiliki 2 nilai yang kosong (NaN). Nilai NaN tersebut bisa muncul akibat proses konversi dengan errors='coerce' atau karena pemetaan sebelumnya tidak mencakup semua nilai yang salah. Informasi ini penting untuk menentukan langkah berikutnya, yaitu apakah kamu ingin mengisi nilai NaN tersebut menggunakan rata-rata, median, atau metode lain sesuai kebutuhan analisis. Ini merupakan bagian dari proses data cleaning agar dataset menjadi bersih dan konsisten sebelum dianalisis lebih lanjut.

In [46]:
ziza_df.head(50)
#untuk memastika bisa menggunakan sintaks di atas dengan masukan value baris yang ingin di lihat, cara ini juga bisa di lakukan untuk tail(50)

nim    nama            jurusan  umur  nilai_akhir  kehadiran_pct  \
0   E0001    Aldi  Pembangkit Energi  25.0         59.5           82.0   
1   E0002    Budi  Pembangkit Energi  20.0         84.5           99.0   
2   E0003   Citra  Pembangkit Energi  25.0         56.2           78.0   
3   E0004    Dewi  Pembangkit Energi  20.0         63.0           99.0   
4   E0005     Eka   Manajemen Energi  22.0        999.0           65.0   
5   E0006   Fajar  Pembangkit Energi  21.0        -20.0           78.0   
6   E0007    Gina        Informatika  19.0         70.2          150.0   
7   E0008  Hendra  Pembangkit Energi  25.0         56.7          -10.0   
8   E0009     Ika        Informatika  25.0         58.2           83.0   
9   E0010    Joko  Pembangkit Energi  23.0         62.7           69.0   
10  E0011    Kiki  Pembangkit Energi  25.0         66.6           70.0   
11  E0012    Lina        Informatika  22.0         66.1           82.0   
12  E0013    Maya   Manajemen Energi  21.0         63.4           82.0   
13  E0014   Nando  Pembangkit Energi  23.0         80.4           99.0   
14  E0015     Oki  Pembangkit Energi  22.0         70.2          100.0   
15  E0016    Putu  Pembangkit Energi  25.0         85.7           68.0   
16  E0017    Qori  Pembangkit Energi  25.0         67.6           82.0   
17  E0018    Rina  Pembangkit Energi  23.0         90.1           85.0   
18  E0019    Sari        Informatika  22.0         90.4           94.0   
19  E0020    Tono        Informatika  20.0         64.9          100.0   
20  E0021     Uli  Pembangkit Energi  24.0         78.4           90.0   
21  E0022    Vina  Pembangkit Energi   NaN         75.4           70.0   
22  E0023   Wawan  Pembangkit Energi  20.0         80.1           92.0   
23  E0024    Xena        Informatika  22.0         70.3           94.0   
24  E0025    Yudi  Pembangkit Energi  23.0         89.4           65.0   
25  E0026    Zara        Informatika  24.0         90.4           82.0   
26  E0027    Arif  Pembangkit Energi  19.0         66.7           75.0   
27  E0028    Bima  Pembangkit Energi  24.0         90.0           81.0   
28  E0029    Cici  Pembangkit Energi  23.0         91.5           84.0   
29  E0030    Dani  Pembangkit Energi  20.0         88.0           75.0   
30  E0031    Eros        Informatika  23.0         68.0           66.0   
31  E0032   Ferry  Pembangkit Energi  25.0         87.3           80.0   
32  E0033  Gilang  Pembangkit Energi  23.0         88.0           70.0   
33  E0034    Hani   Manajemen Energi  25.0         76.3           73.0   
34  E0035   Indra   Manajemen Energi  23.0         61.6           98.0   
35  E0036    Jeni        Informatika  20.0         92.2           77.0   
36  E0037    Kris   Manajemen Energi  22.0         94.8           88.0   
37  E0038    Lulu  Pembangkit Energi  19.0         64.9           69.0   
38  E0039    Mita  Pembangkit Energi  23.0         77.2           79.0   
39  E0040    Nina  Pembangkit Energi  24.0         80.2           79.0   
40  E0041    Omar        Informatika  25.0         88.0           97.0   
41  E0042     Pia   Manajemen Energi  24.0         74.4           99.0   
42  E0043    Rico  Pembangkit Energi  20.0         86.4           91.0   
43  E0044   Sinta   Manajemen Energi  19.0         81.4           87.0   
44  E0045    Tari  Pembangkit Energi  22.0         89.6           68.0   
45  E0046    Umar  Pembangkit Energi  19.0         71.1           86.0   
46  E0047    Vita   Manajemen Energi  20.0         62.6           93.0   
47  E0048    Widi   Manajemen Energi  20.0         66.1           80.0   
48  E0049     Xul  Pembangkit Energi  25.0         89.5          100.0   
49  E0050   Yosef  Pembangkit Energi  24.0         95.0           65.0   

   tanggal_daftar  
0      2024-05-05  
1      2024-02-14  
2      2024-04-29  
3      2024-08-02  
4      2024-03-22  
5      2024-06-21  
6      2024-06-25  
7      2024-10-01  
8      2024-11-17  
9      2024-01-24  
10     05/0

Perintah ziza_df.head(50) digunakan untuk menampilkan 50 baris pertama dari DataFrame sebagai cara untuk melakukan pengecekan visual terhadap data setelah dilakukan pembersihan atau perubahan. Dengan melihat baris-baris teratas, kamu dapat memastikan apakah hasil pemetaan umur, perbaikan jurusan, atau penanganan nilai NaN sudah diterapkan dengan benar. Metode ini membantu menemukan kesalahan lebih cepat dibandingkan hanya melihat ringkasan angka. Selain itu, komentar dalam kode menjelaskan bahwa cara ini juga dapat dilakukan dari arah bawah menggunakan ziza_df.tail(50) jika ingin melihat 50 baris terakhir. Kedua metode ini sangat berguna dalam proses data inspection untuk memastikan seluruh perubahan data sudah sesuai dengan harapan sebelum melanjutkan ke tahap analisis berikutnya.

In [47]:
ziza_df.tail(50)

nim    nama            jurusan  umur  nilai_akhir  kehadiran_pct  \
20  E0021     Uli  Pembangkit Energi  24.0         78.4           90.0   
21  E0022    Vina  Pembangkit Energi   NaN         75.4           70.0   
22  E0023   Wawan  Pembangkit Energi  20.0         80.1           92.0   
23  E0024    Xena        Informatika  22.0         70.3           94.0   
24  E0025    Yudi  Pembangkit Energi  23.0         89.4           65.0   
25  E0026    Zara        Informatika  24.0         90.4           82.0   
26  E0027    Arif  Pembangkit Energi  19.0         66.7           75.0   
27  E0028    Bima  Pembangkit Energi  24.0         90.0           81.0   
28  E0029    Cici  Pembangkit Energi  23.0         91.5           84.0   
29  E0030    Dani  Pembangkit Energi  20.0         88.0           75.0   
30  E0031    Eros        Informatika  23.0         68.0           66.0   
31  E0032   Ferry  Pembangkit Energi  25.0         87.3           80.0   
32  E0033  Gilang  Pembangkit Energi  23.0         88.0           70.0   
33  E0034    Hani   Manajemen Energi  25.0         76.3           73.0   
34  E0035   Indra   Manajemen Energi  23.0         61.6           98.0   
35  E0036    Jeni        Informatika  20.0         92.2           77.0   
36  E0037    Kris   Manajemen Energi  22.0         94.8           88.0   
37  E0038    Lulu  Pembangkit Energi  19.0         64.9           69.0   
38  E0039    Mita  Pembangkit Energi  23.0         77.2           79.0   
39  E0040    Nina  Pembangkit Energi  24.0         80.2           79.0   
40  E0041    Omar        Informatika  25.0         88.0           97.0   
41  E0042     Pia   Manajemen Energi  24.0         74.4           99.0   
42  E0043    Rico  Pembangkit Energi  20.0         86.4           91.0   
43  E0044   Sinta   Manajemen Energi  19.0         81.4           87.0   
44  E0045    Tari  Pembangkit Energi  22.0         89.6           68.0   
45  E0046    Umar  Pembangkit Energi  19.0         71.1           86.0   
46  E0047    Vita   Manajemen Energi  20.0         62.6           93.0   
47  E0048    Widi   Manajemen Energi  20.0         66.1           80.0   
48  E0049     Xul  Pembangkit Energi  25.0         89.5          100.0   
49  E0050   Yosef  Pembangkit Energi  24.0         95.0           65.0   
50  E0051    Zaki   Manajemen Energi  20.0         71.3           95.0   
51  E0052    Adel        Informatika  23.0         61.6           65.0   
52  E0053   Bagus   Manajemen Energi  25.0         86.4           83.0   
53  E0054   Cahyo  Pembangkit Energi   NaN         62.6           78.0   
54  E0055    Dina  Pembangkit Energi  24.0         67.5           68.0   
55  E0056     Edo  Pembangkit Energi  23.0         88.0           98.0   
56  E0057   Fauzi  Pembangkit Energi  23.0         58.2           76.0   
57  E0058    Gita  Pembangkit Energi  24.0         89.5           90.0   
58  E0059    Hadi        Informatika  23.0         78.8           70.0   
59  E0060   Ilham        Informatika  21.0         88.0           85.0   
60  E0061    Jaya        Informatika  22.0         60.2           84.0   
61  E0062  Kirana  Pembangkit Energi  25.0         92.4           65.0   
62  E0063   Lutfi  Pembangkit Energi  19.0         76.5           97.0   
63  E0064    Mega  Pembangkit Energi  21.0         90.2           80.0   
64  E0065   Nadia        Informatika  23.0         72.5           99.0   
65  E0066    Oleh  Pembangkit Energi  23.0         94.4           98.0   
66  E0067   Putri  Pembangkit Energi  24.0         59.1           82.0   
67  E0068    Rafi        Informatika  19.0         84.7           74.0   
68  E0069   Salsa        Informatika  23.0         63.4           86.0   
69  E0070   Tasya   Manajemen Energi  23.0         74.5           68.0   

   tanggal_daftar  
20     2024-07-04  
21     2024-01-25  
22     2024-11-01  
23     2024-09-27  
24     2024-12-14  
25     2024-11-24  
26    24 Nov 2024  
27     2024-09-13  
28     2024-11-23  
29     2024-10-03  
30     2024

Perintah ziza_df.tail(50) digunakan untuk menampilkan 50 baris terakhir dari DataFrame ziza_df. Tujuan utama penggunaan perintah ini adalah untuk melakukan pengecekan visual terhadap bagian akhir dataset setelah proses pembersihan data dilakukan, seperti normalisasi jurusan, pemetaan umur, pengisian missing value, dan konversi tipe data. Dengan melihat baris paling bawah, kamu dapat memastikan bahwa perubahan yang diterapkan konsisten di seluruh dataset, bukan hanya pada baris awal. Selain itu, cara ini berguna untuk mendeteksi apakah masih ada nilai aneh, kosong, atau salah format yang mungkin terlewat pada proses sebelumnya. Secara keseluruhan, tail(50) membantu memastikan kualitas data sebelum dataset digunakan untuk analisis lebih lanjut.

Terdapat Missing value pada baris 21 dan 54

untuk itu bisa di lakukan replace lagi dengan memasukan rata-rata umur seperti sebelumnya

In [48]:
ziza_df['umur'].mean() #mencari rata-rata

np.float64(22.426470588235293)

Perintah ini digunakan untuk menghitung nilai rata-rata dari seluruh data pada kolom umur setelah proses pembersihan dilakukan. Fungsi .mean() akan otomatis mengabaikan nilai NaN, sehingga perhitungan tetap akurat meskipun masih ada data yang kosong. Menghitung rata-rata umur penting dalam proses data cleaning, terutama jika kamu berencana menggunakan nilai rata-rata tersebut untuk menggantikan missing value pada kolom umur. Dengan mengetahui nilai mean ini, kamu bisa memastikan bahwa pengisian nilai yang hilang dilakukan secara konsisten dan berdasarkan representasi umum dari data yang tersedia. Nilai rata-rata juga membantu mengevaluasi apakah ada umur tidak wajar yang masih tersisa di dataset.

In [49]:
ziza_df['umur'] = ziza_df['umur'].fillna(22) #isi dengan nilai rata-rata

Perintah ini digunakan untuk mengisi nilai NaN atau missing value pada kolom umur dengan angka 22, yang merupakan nilai rata-rata umur yang sudah dihitung sebelumnya. Dengan menggunakan .fillna(22), setiap sel yang kosong pada kolom tersebut akan diganti secara otomatis sehingga tidak ada lagi nilai yang hilang. Langkah ini penting agar kolom umur menjadi bersih, lengkap, dan konsisten sehingga bisa digunakan untuk perhitungan statistik atau analisis lebih lanjut tanpa error. Mengisi missing value dengan mean adalah metode yang umum digunakan karena dianggap sebagai representasi yang paling netral dari seluruh distribusi umur yang tersedia dalam dataset.

In [50]:
ziza_df.isna().sum()

nim               0
nama              0
jurusan           0
umur              0
nilai_akhir       0
kehadiran_pct     0
tanggal_daftar    0
dtype: int64

Perintah ini digunakan untuk mengecek kembali jumlah missing value (NaN) pada setiap kolom dalam DataFrame setelah proses pengisian nilai umur dilakukan. Fungsi isna() akan menandai setiap sel yang kosong dengan nilai True, lalu sum() menjumlahkan berapa banyak nilai True di tiap kolom. Dengan menjalankan kode ini, kamu dapat memastikan apakah proses pembersihan data telah berhasil—terutama untuk memeriksa apakah kolom umur sudah tidak memiliki nilai NaN setelah diisi dengan nilai rata-rata. Jika seluruh kolom menunjukkan angka 0, berarti tidak ada lagi data yang hilang dan dataset sudah siap untuk dianalisis lebih lanjut.

Setelah di cek sudah tidak ada lagi data yang missing pada semua kolom data

In [51]:
ziza_df.duplicated().sum()


np.int64(0)

Perintah ini digunakan untuk mengetahui berapa banyak baris duplikat yang ada di dalam DataFrame ziza_df. Fungsi .duplicated() akan mengecek setiap baris dan menandai baris yang isinya sama persis dengan baris sebelumnya sebagai True, kemudian .sum() menghitung total nilai True tersebut. Dengan cara ini, kamu dapat mengetahui jumlah data yang terduplikasi tanpa langsung menghapusnya. Informasi ini penting dalam proses data cleaning, karena data duplikat dapat menyebabkan hasil analisis menjadi bias atau tidak akurat. Jika hasilnya lebih dari nol, berarti masih ada baris data yang perlu diperiksa dan kemungkinan besar dihapus menggunakan drop_duplicates() untuk memastikan dataset tetap bersih dan valid.

Untuk pulikat juga sama tidak ada data yang duplikat

Untuk memverifikasi bisa show head() dan tail() seperti di bawah ini atau count duplicates() dan isnan()

In [52]:
ziza_df.head(55)

nim    nama            jurusan  umur  nilai_akhir  kehadiran_pct  \
0   E0001    Aldi  Pembangkit Energi  25.0         59.5           82.0   
1   E0002    Budi  Pembangkit Energi  20.0         84.5           99.0   
2   E0003   Citra  Pembangkit Energi  25.0         56.2           78.0   
3   E0004    Dewi  Pembangkit Energi  20.0         63.0           99.0   
4   E0005     Eka   Manajemen Energi  22.0        999.0           65.0   
5   E0006   Fajar  Pembangkit Energi  21.0        -20.0           78.0   
6   E0007    Gina        Informatika  19.0         70.2          150.0   
7   E0008  Hendra  Pembangkit Energi  25.0         56.7          -10.0   
8   E0009     Ika        Informatika  25.0         58.2           83.0   
9   E0010    Joko  Pembangkit Energi  23.0         62.7           69.0   
10  E0011    Kiki  Pembangkit Energi  25.0         66.6           70.0   
11  E0012    Lina        Informatika  22.0         66.1           82.0   
12  E0013    Maya   Manajemen Energi  21.0         63.4           82.0   
13  E0014   Nando  Pembangkit Energi  23.0         80.4           99.0   
14  E0015     Oki  Pembangkit Energi  22.0         70.2          100.0   
15  E0016    Putu  Pembangkit Energi  25.0         85.7           68.0   
16  E0017    Qori  Pembangkit Energi  25.0         67.6           82.0   
17  E0018    Rina  Pembangkit Energi  23.0         90.1           85.0   
18  E0019    Sari        Informatika  22.0         90.4           94.0   
19  E0020    Tono        Informatika  20.0         64.9          100.0   
20  E0021     Uli  Pembangkit Energi  24.0         78.4           90.0   
21  E0022    Vina  Pembangkit Energi  22.0         75.4           70.0   
22  E0023   Wawan  Pembangkit Energi  20.0         80.1           92.0   
23  E0024    Xena        Informatika  22.0         70.3           94.0   
24  E0025    Yudi  Pembangkit Energi  23.0         89.4           65.0   
25  E0026    Zara        Informatika  24.0         90.4           82.0   
26  E0027    Arif  Pembangkit Energi  19.0         66.7           75.0   
27  E0028    Bima  Pembangkit Energi  24.0         90.0           81.0   
28  E0029    Cici  Pembangkit Energi  23.0         91.5           84.0   
29  E0030    Dani  Pembangkit Energi  20.0         88.0           75.0   
30  E0031    Eros        Informatika  23.0         68.0           66.0   
31  E0032   Ferry  Pembangkit Energi  25.0         87.3           80.0   
32  E0033  Gilang  Pembangkit Energi  23.0         88.0           70.0   
33  E0034    Hani   Manajemen Energi  25.0         76.3           73.0   
34  E0035   Indra   Manajemen Energi  23.0         61.6           98.0   
35  E0036    Jeni        Informatika  20.0         92.2           77.0   
36  E0037    Kris   Manajemen Energi  22.0         94.8           88.0   
37  E0038    Lulu  Pembangkit Energi  19.0         64.9           69.0   
38  E0039    Mita  Pembangkit Energi  23.0         77.2           79.0   
39  E0040    Nina  Pembangkit Energi  24.0         80.2           79.0   
40  E0041    Omar        Informatika  25.0         88.0           97.0   
41  E0042     Pia   Manajemen Energi  24.0         74.4           99.0   
42  E0043    Rico  Pembangkit Energi  20.0         86.4           91.0   
43  E0044   Sinta   Manajemen Energi  19.0         81.4           87.0   
44  E0045    Tari  Pembangkit Energi  22.0         89.6           68.0   
45  E0046    Umar  Pembangkit Energi  19.0         71.1           86.0   
46  E0047    Vita   Manajemen Energi  20.0         62.6           93.0   
47  E0048    Widi   Manajemen Energi  20.0         66.1           80.0   
48  E0049     Xul  Pembangkit Energi  25.0         89.5          100.0   
49  E0050   Yosef  Pembangkit Energi  24.0         95.0           65.0   
50  E0051    Zaki   Manajemen Energi  20.0         71.3           95.0   
51  E0052    Adel        Informatika  23.0         61.6           65.0   
52  E0053   Bagus   Manajemen Energi  25.0         86.4           83.0   
53  E0054 

Perintah ziza_df.head(55) dalam Python digunakan untuk menampilkan 55 baris pertama dari DataFrame bernama ziza_df. Fungsi .head() ini sangat berguna untuk melakukan pengecekan awal pada dataset, misalnya untuk melihat struktur data, kolom, dan beberapa nilai awal tanpa harus menampilkan seluruh isi DataFrame. Jika jumlah baris dalam DataFrame kurang dari 55, maka Python akan menampilkan semua baris yang tersedia. Perlu dicatat bahwa penggunaan .head() hanya menampilkan data dan tidak mengubah isi DataFrame itu sendiri. Sebagai alternatif, jika ingin melihat baris terakhir dari DataFrame, dapat menggunakan fungsi .tail(n) dengan jumlah baris yang diinginkan. Dengan cara ini, kita bisa lebih mudah memahami dan memeriksa data sebelum melakukan analisis lebih lanjut.

In [53]:
ziza_df.tail()

nim   nama            jurusan  umur  nilai_akhir  kehadiran_pct  \
65  E0066   Oleh  Pembangkit Energi  23.0         94.4           98.0   
66  E0067  Putri  Pembangkit Energi  24.0         59.1           82.0   
67  E0068   Rafi        Informatika  19.0         84.7           74.0   
68  E0069  Salsa        Informatika  23.0         63.4           86.0   
69  E0070  Tasya   Manajemen Energi  23.0         74.5           68.0   

   tanggal_daftar  
65     2024-01-05  
66     2024-05-15  
67     2024-05-19  
68     2024-04-14  
69     2024-02-17

Perintah ziza_df.tail() dalam Python digunakan untuk menampilkan 5 baris terakhir dari DataFrame ziza_df secara default. Fungsi .tail() sangat berguna untuk memeriksa data di bagian akhir dataset, misalnya untuk melihat apakah ada nilai yang hilang, duplikat, atau pola tertentu yang muncul di baris terakhir. Sama seperti .head(), penggunaan .tail() hanya menampilkan data dan tidak mengubah isi DataFrame. Jika ingin menampilkan jumlah baris yang berbeda, misalnya 10 atau 20 baris terakhir, cukup menambahkan angka sebagai argumen, misalnya ziza_df.tail(10). Dengan cara ini, kita bisa mendapatkan gambaran lengkap dari kedua ujung dataset sebelum melakukan analisis lebih lanjut.

In [54]:
ziza_df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 70 entries, 0 to 69
Data columns (total 7 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   nim             70 non-null     object 
 1   nama            70 non-null     object 
 2   jurusan         70 non-null     object 
 3   umur            70 non-null     float64
 4   nilai_akhir     70 non-null     float64
 5   kehadiran_pct   70 non-null     float64
 6   tanggal_daftar  70 non-null     object 
dtypes: float64(3), object(4)
memory usage: 4.4+ KB


Perintah ziza_df.info() dalam Python digunakan untuk menampilkan ringkasan informasi tentang DataFrame ziza_df. Ketika dijalankan, fungsi ini akan memberikan beberapa informasi penting, antara lain:

Jumlah total baris dalam DataFrame.

Nama setiap kolom beserta tipe data (dtype) masing-masing, seperti int64, float64, atau object.

Jumlah nilai non-null pada setiap kolom, sehingga kita bisa cepat mengetahui kolom mana yang memiliki nilai kosong atau missing data.

Penggunaan memori oleh DataFrame, berguna untuk memperkirakan efisiensi penyimpanan data, terutama pada dataset besar.

Fungsi .info() sangat berguna untuk memahami struktur dataset secara cepat sebelum melakukan analisis atau pembersihan data lebih lanjut. Ini berbeda dengan .head() atau .tail() yang hanya menampilkan sebagian data; .info() memberikan gambaran keseluruhan tentang kolom dan tipe datanya.

JIka di lihat pada info, maka semua jumlah data sama yaitu 70, maka proses Pembersihan data sudah selesai

Export data yang sudah di bersihkan menjadi file csv, menggunakan script berikut ini

In [55]:
ziza_df.to_csv("/content/mahasiswa_clean.csv", index=False)

Perintah ziza_df.to_csv("/content/mahasiswa_clean.csv", index=False) digunakan untuk menyimpan DataFrame ziza_df ke dalam file CSV baru bernama mahasiswa_clean.csv di folder /content/. Beberapa hal penting tentang perintah ini:

index=False memastikan nomor indeks DataFrame tidak ikut tersimpan di file CSV. Jika tidak ditulis, kolom indeks default (0,1,2,…) akan ikut masuk ke file CSV.

File yang tersimpan dapat dibuka di Excel, Google Sheets, atau dibaca kembali di Python menggunakan pd.read_csv().

Ini biasanya dilakukan setelah membersihkan data atau memodifikasi DataFrame, sehingga file baru berisi data yang sudah bersih atau diolah.

Setelah perintah ini dijalankan, file mahasiswa_clean.csv siap digunakan untuk analisis selanjutnya atau dibagikan tanpa mengubah file CSV asli.

Lalu lakukan Dodownload file csv yang sudah di export sebelumnya

In [56]:
from google.colab import files
files.download("/content/mahasiswa_clean.csv")

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

Bandingkan hasil data csv sebelumnya dengan csv yang sudah di bersihkan bisa di buka menggunakan excel / spreadsheet untuk membandingaknya

kan di jelskan langkah-langkah dan hasil latihan pada laporan